homeassistant/components/tplink/camera.py

```diff
@@ -0,0 +1,221 @@
+        if (camera_module := device.modules.get(Module.Camera)) and live_view
+    )
+
+
+class TPLinkCameraEntity(CoordinatedTPLinkEntity, Camera):
+    """Representation of a TPLink camera."""
+
+    IMAGE_INTERVAL = datetime.timedelta(minutes=5)
+
+    _attr_supported_features = CameraEntityFeature.STREAM | CameraEntityFeature.ON_OFF
+
+    entity_description: TPLinkCameraEntityDescription
+
+    def __init__(
+        self,
+        device: Device,
+        coordinator: TPLinkDataUpdateCoordinator,
+        description: TPLinkCameraEntityDescription,
+        *,
+        camera_module: CameraModule,
+        parent: Device | None = None,
+        ffmpeg_manager: ffmpeg.FFmpegManager,
+        camera_credentials: Credentials | None,
+    ) -> None:
+        """Initialize a TPlink camera."""
+        self.entity_description = description
+        self._camera_module = camera_module
+        self._video_url = camera_module.stream_rtsp_url(camera_credentials)
+        self._image: bytes | None = None
+        super().__init__(device, coordinator, parent=parent)
+        Camera.__init__(self)
+        self._ffmpeg_manager = ffmpeg_manager
+        self._image_lock = asyncio.Lock()
+        self._last_update = datetime.datetime.min
+        self._camera_credentials = camera_credentials
+        self._can_stream = True
+        self._http_mpeg_stream_running = False
+
+    def _get_unique_id(self) -> str:
+        """Return unique ID for the entity."""
+        return f"{legacy_device_id(self._device)}-{self.entity_description}"
+
+    @callback
+    def _async_update_attrs(self) -> None:
+        """Update the entity's attributes."""
+        self._attr_is_on = self._camera_module.is_on
+
+    async def stream_source(self) -> str | None:
+        """Return the source of the stream."""
+        return self._video_url
+
+    async def _async_check_stream_auth(self, video_url: str) -> None:
+        """Check for an auth error and start reauth flow."""
+        try:
+            await stream.async_check_stream_client_error(self.hass, video_url)
+        except stream.StreamOpenClientError as ex:
+            if ex.stream_client_error is stream.StreamClientError.Unauthorized:
+                _LOGGER.debug(
+                    "Camera stream failed authentication for %s",
+                    self._device.host,
+                )
+                self._can_stream = False
+                self.coordinator.config_entry.async_start_reauth(
+                    self.hass,
+                    ConfigFlowContext(
+                        reauth_source=CONF_CAMERA_CREDENTIALS,  # type: ignore[typeddict-unknown-key]
+                    ),
+                    {"device": self._device},
+                )
+
+    async def async_camera_image(
+        self, width: int | None = None, height: int | None = None
+    ) -> bytes | None:
+        """Return a still image response from the camera."""
+        now = dt_util.utcnow()
```

Comment:
Its usually better to use monotonic time vs clock time since it can't move backwards after ntp

Reply:
That's done